### PR TITLE
Best practices for developer docs

### LINKED ISSUE #339: [DOC] Publish Best practices for developers documentation
## Issue description

- Convert '[Best practices for developer documentation](https://techweb.axway.com/confluence/display/RDAPI/Best+practices+for+developer+documentation)' from Confluence to Markdown
- Publish the best practices under the Contribution Guidelines
- Share it with TWs Guild.

### PR DESCRIPTION
**Pull request by andreamussap**
Thank you for your contribution to the Axway-Open-Docs repo.

## Describe the changes

Creating Best practices for developer docs. Fixes #339 

## Checklist for contributors

Before submitting this PR, please make sure:

* [x] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [x] You have signed the [Axway CLA](https://cla.axway.com/)
* [x] You have verified the technical accuracy of your change
* [x] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)

_Put an x in the boxes that apply. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your change._

## Checklist for approvers

_This section is to be filled out by project maintainers only._

Before approving this PR, please make sure it:

* [x] Does not have conflicts with the base branch
* [x] Is clear and complete
* [x] Is believed to be accurate (technical accuracy to be checked with an SME for PRs originating from outside R&D)
* [x] Follows [Axway style](https://techweb.axway.com/confluence/display/RIE/Style+guide)
* [x] Follows [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)
* [x] Follows [best practices](https://techweb.axway.com/confluence/display/RDAPI/Best+practices+for+developer+documentation)
* [x] Does not contain typos, grammatical errors, etc.
* [x] Passes the Markdown linter rules (automated via CI check)
* [x] Does not contain broken links
* [x] Is discoverable and navigable

## Checklist for mergers

_This section is to be filled out by project maintainers only._

Before merging this PR, please make sure:

* [ ] You have applied a size label (unless this work is tracked in the ReadMeFirst JIRA project)
* [ ] You have applied a production label if this needs a manual push to Zoomin
* [ ] You have added it to the correct GitHub project
* [ ] You have added it to the correct Sprint milestone
* [ ] You have appended a `fixes` line to this description if this PR fixes a specific GitHub issue

### CODE REVIEW

**Comment by andreamussap** [comment]
@alexearnshaw This is ready for review

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+Old, outdated, non-used guides can misinform and slow down the user experience on the website. They can show results in the search, which are not relevant (or, even correct) anymore.
+
+### Do not duplicate information
+
+Duplicated content creates inconsistency and frustration, and are very hard to maintain.
+
+* Avoid duplication of information on your own guides
+* Record the information once where it enhances the work the most
+* Do not write your own guide to a common technology or process. Link to it instead
```
Comment (alexearnshaw) [inline content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: This list item is 2 sentences so needs a period at the end as well. All other list items in this list then also need ending periods for consistency

Comment (alexearnshaw) [reply content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: Or you could just rewrite this list item as one sentence:
Do not write your own guide to a common technology or process. Link to it instead => Link to information about a common technology or process instead of rewriting the information in your own words

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+## Writing style
+
+Follow these guidelines to make your content concise and clearer.
```
Comment (alexearnshaw) [inline content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: concise and clear OR more concise and clearer

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+### Page length
+
+* Determine page length by relevance of information
```
Comment (alexearnshaw) [inline content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: This is not clear to me? How does relevance of information determine page length?

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+* it is ok to have long pages when the informational units are related. Users don't have to scroll down the page, they can use the right-side menu (available on Axway open docs pages) or they can search (Ctrl + F) for what they're looking for
```
Comment (alexearnshaw) [inline content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: Longer pages are useful when the informational units are related. Users don't have to scroll down the page, and can use the right-side navigation menu, or they can search (Ctrl + F) the page.

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+* Long pages saves people from jumping back and forth between pages when the related and relevant information could be all in one place
```
Comment (alexearnshaw) [inline content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: Longer pages reduce the need to jump back and forth between pages when the related and relevant information could be all in one place.

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+### Place the information in the most appropriate place
+
+Assess the best place for chunks of information by the needs of the customer of that information, where are they most likely to need it.
+
+Think about some questions to figure out what is the best place for the content:
```
Comment (alexearnshaw) [inline content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: Most of these questions don't seem relevant to our kind of docs? I think either remove the questions or come up with questions that are more relevant for our user documentation

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+* Remember that users scan in F-Shaped pattern
+* Break up the text often
+* First 3-5 words of every paragraph are the most important
+* Use bullet lists
+* Use variations in typeface (links, bold, etc)
+* Use code snippets
+* Use screenshots sparingly
+* Use animated gifs
+
+## Using hyperlinks
```
Comment (alexearnshaw) [inline content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: Using => Use meaningful link text?

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+
+Readers use links as guideposts when scanning. Effective link text helps readers to understand where each link leads and decide whether they want to click the link or not.
+
+Observe these guidelines to improve user experience and accessibility.
+
+* Do not paste the actual URL in the page as this does not read well, specially for screen readers (accessibility)
+* Do not use **Click here**. it is bad for SEO, and does not describe the target. Provide a meaningful description, label to the link indicating what the user will see after following the link
```
Comment (alexearnshaw) [inline content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: Suggest splitting this item into 2 list items

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+* Choose keywords or phrases that best represent the content of the destination
+
+## Do not overuse tables
+
+In general, you should use tables to display data as a way to make your document or web page look the way you want it to.
```
Comment (alexearnshaw) [inline content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: as a way to make your document or web page look the way you want it to.
I don't understand this? How does a table make a page look the way you want???

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+
+Not all analyses or results warrant a table or image. Some simple results are best stated in a simple and short paragraph.
+
+Tables can be used as quick references and can reveal trends, patterns, or relationships that might otherwise be difficult to grasp in plain-text format. They permit rapid access to and relatively easy comparison of information.
+
+Contrary, overly complicated tables may be difficult to understand, so strive for simplicity (single-cell tables). Complex tables can be a challenge for adaptive technology users (users of screen reader software).
```
Comment (alexearnshaw) [inline content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: Contrary => In comparison is more usual

Comment (alexearnshaw) [reply content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: What's a single-cell table? Do you mean where each cell contains only 1 piece of information?

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+
+Accessible tables are simple, and they have an identified header row, and include a table summary, either as a caption or as alt text. These techniques help screen reader users read the information contained in the table.
+
+## Do not overuse screenshots
+
+Screenshots are hard to maintain, and if they aren't up to date they can be misleading.
+
+Too many unnecessary screenshots can also clutter the page.
+
+If you can present your results clearly in a few short sentences, or in a list of steps, this means that an image is probably unnecessary.
```
Comment (alexearnshaw) [inline content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: Although the video guidelines need more research, I think it's worth adding them here as a general guide e.g.
Videos should not replace documentation, but complement or illustrate it. 

Also I think we must include the list of reference articles in this topic, especially as people will want to know where we got these BPs from, and also as we have copied some of their text, listing them as refs avoids any plagiarism concerns.

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md [hunk 12]
```diff
@@ -0,0 +1,191 @@
+---
+title: "Best practices for developer documentation"
+linkTitle: "Best practices for developer documentation"
```
[the remark this answers is not in the document]
Comment (andreamussap) [reply content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: @alexearnshaw  I don't think that 'best practices' on its own is meaningful, 'Best practices' for what? could be for anything ??

[the remark this answers is not in the document]
Comment (alexearnshaw) [reply content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: This topic is under contrib guidelines so for me that implies it's something to do with writing or contributing documentation. We need to keep our link titles as short as possible for easy navigation, the title field should still stay as "Best practices for developer documentation"

File: content/en/docs/contribution_guidelines/bestpracticedevdoc.md (hunk 6 above)
Comment (andreamussap) [reply content/en/docs/contribution_guidelines/bestpracticedevdoc.md]: I've removed this section instead because since this guide if for all Axway products, 'the most appropriate place' can be relative, so it's hard to come up with the right/ specific questions.

**Comment by andreamussap** [comment]
> @andreamussap review is done, I've added some suggestions for improvement and clarity, let me know if you have questions.

@alexearnshaw thank you for fback. I also have made some other improvements to the text. It's read for review again. Thanks

**Comment by andreamussap** [comment]
@alexearnshaw  when this is merged I think I should delete the [confluence page](https://techweb.axway.com/confluence/display/RDAPI/Best+practices+for+developer+documentation) to avoid duplication of information. Agree?

**Comment by alexearnshaw** [comment]
> @alexearnshaw when this is merged I think I should delete the [confluence page](https://techweb.axway.com/confluence/display/RDAPI/Best+practices+for+developer+documentation) to avoid duplication of information. Agree?

@andreamussap This looks great now, thanks! 

Yes, agree with removing Confluence page. Can you also update the PR template with the new link to the BPs (as it currently links to Confluence)
